core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -384,59 +394,49 @@ Subchannel getSubchannel(String stickinessValue) {
       }
     }
   }
+  
+  // Only subclasses are ReadyPicker or EmptyPicker
+  private abstract static class RoundRobinPicker extends SubchannelPicker {
+    abstract boolean isEquivalentTo(RoundRobinPicker picker);
+  }
 
   @VisibleForTesting
-  static final class Picker extends SubchannelPicker {
-    private static final AtomicIntegerFieldUpdater<Picker> indexUpdater =
-        AtomicIntegerFieldUpdater.newUpdater(Picker.class, "index");
+  static final class ReadyPicker extends RoundRobinPicker {
+    private static final AtomicIntegerFieldUpdater<ReadyPicker> indexUpdater =
+        AtomicIntegerFieldUpdater.newUpdater(ReadyPicker.class, "index");
 
-    @Nullable
-    private final Status status;
-    private final List<Subchannel> list;
+    private final List<Subchannel> list; // non-empty
     @Nullable
     private final RoundRobinLoadBalancer.StickinessState stickinessState;
     @SuppressWarnings("unused")
     private volatile int index;
 
-    Picker(
-        List<Subchannel> list, @Nullable Status status, int startIndex,
+    ReadyPicker(List<Subchannel> list, int startIndex,
         @Nullable RoundRobinLoadBalancer.StickinessState stickinessState) {
+      assert !list.isEmpty();
```

Comment:
This should be a checkArgument(). If this did happen in production and it was let pass this point, it would cause an out-of-bound exception at a later point.